lib/patch/functions.js

```diff
@@ -48,6 +48,49 @@ function patchSetClearFunction(obj, fnNames) {
   });
 };
 
+
+/**
+ * Patch requestAnimationFrame.  This function is typically recursively called from within
+ * the callback function that it executes.
+ *
+ *  1) If the patched function is called recursively, bind to the
+ *     parent zone and increment the reference count.
+ *  2) After the callback function has executed, decrement the reference count.
+ *  3) If the reference count is <= 0, dequeue the current task.
+ */
+function patchRequestAnimationFrame(obj, fnNames) {
+  fnNames.forEach(function (name) {
+    var delegate = obj[name];
+    if (delegate) {
+      global.zone[name] = function (fn) {
+        if (fn) {
```

Comment:
```js
        if (fn) {
          // vicb: this would be called first, moved on top
          if (typeof fn._zonejs_refs === 'undefined') {
            fn._zonejs_refs = 1;
            fn._zonejs_zone = global.zone;       
          } else {
            fn._zonejs_refs++;
          }

          arguments[0] = function () {
            // vicb: moved up (in case the fn throws)
            if (--fn._zonejs_refs === 0) {
              delete fn._zonejs_refs;
            }

            fn.apply(this, arguments);
          };

          // vicb: by using bindOnce dequeue should not be required any more
          arguments[0] = fn._zonejs_zone.bindOnce(arguments[0]);
        }
```

Notes:
- changed the property name to make collision less likely,
- moved up some code so that it match execution order,
- added some comments to explain the changes (should be removed).

Feel free to use it or now